Failed to build wasmer/examples/echo-server

Can't build `wasmer/examples/echo-server`. Same error for `wasmer/examples/http-server`.
Rust: `rustc 1.44.0-nightly`

```
wasmer/examples/echo-server$ cargo build --target=wasm32-wasi
   Compiling kernel-net v0.1.0 (/home/pc/dev/rs/wasmer/lib/kernel-net)
warning: trait objects without an explicit `dyn` are deprecated
  --> /home/pc/dev/rs/wasmer/lib/kernel-net/src/lib.rs:62:26
   |
62 |     callback: Option<Box<FnOnce()>>,
   |                          ^^^^^^^^ help: use `dyn`: `dyn FnOnce()`
   |
   = note: `#[warn(bare_trait_objects)]` on by default

warning: trait objects without an explicit `dyn` are deprecated
  --> /home/pc/dev/rs/wasmer/lib/kernel-net/src/lib.rs:68:30
   |
68 |     imm_queue: Mutex<Vec<Box<FnOnce()>>>,
   |                              ^^^^^^^^ help: use `dyn`: `dyn FnOnce()`

warning: trait objects without an explicit `dyn` are deprecated
   --> /home/pc/dev/rs/wasmer/lib/kernel-net/src/lib.rs:166:10
    |
166 | ) -> Box<FnOnce()> {
    |          ^^^^^^^^ help: use `dyn`: `dyn FnOnce()`

warning: trait objects without an explicit `dyn` are deprecated
   --> /home/pc/dev/rs/wasmer/lib/kernel-net/src/lib.rs:181:10
    |
181 | ) -> Box<FnOnce()> {
    |          ^^^^^^^^ help: use `dyn`: `dyn FnOnce()`

error[E0277]: the trait bound `InAddr: std::default::Default` is not satisfied
   --> /home/pc/dev/rs/wasmer/lib/kernel-net/src/lib.rs:228:5
    |
228 |     sin_addr: InAddr,
    |     ^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `InAddr`
    |
    = note: required by `std::default::Default::default`
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `kernel-net`.

```

--- a/.azure/install-rust.yml
+++ b/.azure/install-rust.yml
@@ -44,6 +44,7 @@ steps:
       rustup update --no-self-update $RUST_TOOLCHAIN
       rustup default $RUST_TOOLCHAIN
       rustup target add x86_64-unknown-linux-musl
+      rustup target add wasm32-wasi
 
       if [ -n "$ANDROID" ]; then
         rustup target add x86_64-linux-android --toolchain $RUST_TOOLCHAIN

--- a/Makefile
+++ b/Makefile
@@ -207,11 +207,17 @@ check-bench-llvm:
 
 check-bench: check-bench-singlepass check-bench-llvm
 
+check-kernel-net:
+	cargo check -p kernel-net --target=wasm32-wasi
+
+# checks that require a nightly version of Rust
+check-nightly: check-kernel-net
+
 # TODO: We wanted `--workspace --exclude wasmer-runtime`, but can't due
 # to https://github.com/rust-lang/cargo/issues/6745 .
 NOT_RUNTIME_CRATES = -p wasmer-clif-backend -p wasmer-singlepass-backend -p wasmer-middleware-common -p wasmer-runtime-core -p wasmer-emscripten -p wasmer-llvm-backend -p wasmer-wasi -p wasmer-kernel-loader -p wasmer-interface-types
 RUNTIME_CHECK = cargo check --manifest-path lib/runtime/Cargo.toml --no-default-features
-check: check-bench check-kernel-net
+check: check-bench
 	cargo check $(NOT_RUNTIME_CRATES)
 	cargo check --release $(NOT_RUNTIME_CRATES)
 	cargo check --all-features $(NOT_RUNTIME_CRATES)
@@ -262,8 +268,6 @@ check: check-bench check-kernel-net
 	$(RUNTIME_CHECK) --release \
 		--features=llvm,default-backend-llvm
 		--features=default-backend-singlepass,singlepass,cranelift,llvm,cache,deterministic-execution
-
-check-kernel-net: cargo +nightly check -p kernel-net --target=wasm32-wasi
 
 # Release
 release:

--- a/Makefile
+++ b/Makefile
@@ -211,7 +211,7 @@ check-bench: check-bench-singlepass check-bench-llvm
 # to https://github.com/rust-lang/cargo/issues/6745 .
 NOT_RUNTIME_CRATES = -p wasmer-clif-backend -p wasmer-singlepass-backend -p wasmer-middleware-common -p wasmer-runtime-core -p wasmer-emscripten -p wasmer-llvm-backend -p wasmer-wasi -p wasmer-kernel-loader -p wasmer-interface-types
 RUNTIME_CHECK = cargo check --manifest-path lib/runtime/Cargo.toml --no-default-features
-check: check-bench
+check: check-bench check-kernel-net
 	cargo check $(NOT_RUNTIME_CRATES)
 	cargo check --release $(NOT_RUNTIME_CRATES)
 	cargo check --all-features $(NOT_RUNTIME_CRATES)
@@ -262,6 +262,8 @@ check: check-bench
 	$(RUNTIME_CHECK) --release \
 		--features=llvm,default-backend-llvm
 		--features=default-backend-singlepass,singlepass,cranelift,llvm,cache,deterministic-execution
+
+check-kernel-net: cargo +nightly check -p kernel-net --target=wasm32-wasi
 
 # Release
 release:

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -116,7 +116,9 @@ jobs:
       - template: .azure/install-llvm.yml
       - template: .azure/install-sccache.yml
       - template: .azure/install-cmake.yml
-      - bash: make check
+      - bash: |
+          make check
+          make check-nightly
         displayName: Check with Flags
         condition: and(succeeded(), not(eq(variables['Agent.OS'], 'Windows_NT')))
 

--- a/lib/kernel-net/src/lib.rs
+++ b/lib/kernel-net/src/lib.rs
@@ -59,13 +59,13 @@ thread_local! {
 
 #[derive(Default)]
 struct AsyncState {
-    callback: Option<Box<FnOnce()>>,
+    callback: Option<Box<dyn FnOnce()>>,
     _epoll: Option<Arc<Epoll>>,
 }
 
 pub struct Epoll {
     fd: i32,
-    imm_queue: Mutex<Vec<Box<FnOnce()>>>,
+    imm_queue: Mutex<Vec<Box<dyn FnOnce()>>>,
 }
 
 impl Epoll {
@@ -163,7 +163,7 @@ fn get_async_io_payload<
     direction: EpollDirection,
     poll_action: P,
     on_ready: F,
-) -> Box<FnOnce()> {
+) -> Box<dyn FnOnce()> {
     __get_async_io_payload(epoll, fd, direction, poll_action, on_ready, false)
 }
 
@@ -178,7 +178,7 @@ fn __get_async_io_payload<
     mut poll_action: P,
     on_ready: F,
     registered: bool,
-) -> Box<FnOnce()> {
+) -> Box<dyn FnOnce()> {
     let epfd = epoll.fd;
     Box::new(move || {
         //println!("async io payload");
@@ -230,7 +230,7 @@ struct SockaddrIn {
 }
 
 #[repr(C)]
-#[derive(Copy, Clone)]
+#[derive(Default, Copy, Clone)]
 struct InAddr {
     s_addr: u32,
 }

--- a/lib/runtime-core/src/memory/ptr.rs
+++ b/lib/runtime-core/src/memory/ptr.rs
@@ -301,7 +301,7 @@ mod test {
         assert!(unsafe { end_wasm_ptr_array.deref_mut(&memory, 0, 1).is_some() });
         let invalid_idx_len_combos: [(u32, u32); 3] =
             [(last_valid_address_for_u8 + 1, 0), (0, 2), (1, 1)];
-        for &(idx, len) in invalid_idx_len_combos.into_iter() {
+        for &(idx, len) in invalid_idx_len_combos.iter() {
             assert!(end_wasm_ptr_array.deref(&memory, idx, len).is_none());
             assert!(unsafe { end_wasm_ptr_array.deref_mut(&memory, idx, len).is_none() });
         }
@@ -322,7 +322,7 @@ mod test {
             WasmPtr::new(last_valid_address_for_u32 + 3),
             WasmPtr::new(last_valid_address_for_u32 + 4),
         ];
-        for oob_end_ptr in end_wasm_ptr_oob_array.into_iter() {
+        for oob_end_ptr in end_wasm_ptr_oob_array.iter() {
             assert!(oob_end_ptr.deref(&memory).is_none());
             assert!(unsafe { oob_end_ptr.deref_mut(&memory).is_none() });
         }
@@ -332,7 +332,7 @@ mod test {
 
         let invalid_idx_len_combos: [(u32, u32); 3] =
             [(last_valid_address_for_u32 + 1, 0), (0, 2), (1, 1)];
-        for &(idx, len) in invalid_idx_len_combos.into_iter() {
+        for &(idx, len) in invalid_idx_len_combos.iter() {
             assert!(end_wasm_ptr_array.deref(&memory, idx, len).is_none());
             assert!(unsafe { end_wasm_ptr_array.deref_mut(&memory, idx, len).is_none() });
         }
@@ -344,7 +344,7 @@ mod test {
             WasmPtr::new(last_valid_address_for_u32 + 4),
         ];
 
-        for oob_end_array_ptr in end_wasm_ptr_array_oob_array.into_iter() {
+        for oob_end_array_ptr in end_wasm_ptr_array_oob_array.iter() {
             assert!(oob_end_array_ptr.deref(&memory, 0, 1).is_none());
             assert!(unsafe { oob_end_array_ptr.deref_mut(&memory, 0, 1).is_none() });
             assert!(oob_end_array_ptr.deref(&memory, 1, 0).is_none());

